shadows/framework/src/main/java/org/robolectric/shadows/ShadowApplicationPackageManager.java

```diff
@@ -280,22 +280,39 @@ protected PackageInfo getPackageInfo(String packageName, int flags) throws NameN
     // Create a copy of the list for filtering
     resolveInfoList = new ArrayList<>(resolveInfoList);
 
-    if ((flags & PackageManager.MATCH_SYSTEM_ONLY) == PackageManager.MATCH_SYSTEM_ONLY) {
```

Comment:
Might be worth extracting a method here

isFlagSet(int flags, int flag) - your call.